### PR TITLE
code clean up for batching.

### PR DESCRIPTION
**Pull request by freemabd1**
Cleaned up the code and fixed test cases. Also add rollback when exceptions happen.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java [hunk 1]
```diff
@@ -110,80 +112,58 @@ public CohortReview saveCohortReview(CohortReview cohortReview) {
...
 
         try {
             connection = jdbcTemplate.getDataSource().getConnection();
             statement = connection.createStatement();
-            connection.setAutoCommit(true);
+            connection.setAutoCommit(false);
 
             for (ParticipantCohortStatus pcs : participantCohortStatuses) {
-                sqlStatement = StringUtils.isEmpty(sqlStatement) ? sqlTemplate : sqlStatement + nextInsert;
+                sqlStatement = StringUtils.isEmpty(sqlStatement) ? SQL_TEMPLATE : sqlStatement + NEXT_INSERT;
```
Comment (danqrodney) [inline api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java]: I would change your code to call String.format() on just the NEXT_INSERT bit. Formatting an ever-growing string is going to be much slower.

Nit: You could just start with sqlStatement set to SQL_TEMPLATE outside the loop and below, and get rid of EMPTY_STRING.

File: api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java [hunk 2]
```diff
@@ -209,4 +189,37 @@ public ParticipantCohortStatus findParticipantCohortStatus(Long cohortReviewId,
     public Slice<ParticipantCohortStatus> findParticipantCohortStatuses(Long cohortReviewId, PageRequest pageRequest) {
         return participantCohortStatusDao.findByParticipantKey_CohortReviewId(cohortReviewId, pageRequest);
     }
+
+    private void close(Connection connection) {
+        if (connection != null) {
+            try {
+                connection.close();
```
Comment (danqrodney) [inline api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java]: You may want to note here that this doesn't actually close the underlying pooled connection... but is the right thing to do anyway... I think. :)

File: api/src/test/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImplTest.java [hunk 3]
```diff
@@ -200,60 +200,97 @@ public void saveCohortReview() throws Exception {
...
+        participantCohortStatus.setRaceConceptId(1L);
+        participantCohortStatus.setBirthDate(new Date(System.currentTimeMillis()));
+        pcsList.add(participantCohortStatus);
+
+        String sqlStatement = "insert into participant_cohort_status(" +
+                "birth_date, ethnicity_concept_id, gender_concept_id, race_concept_id, " +
+                "status, cohort_review_id, participant_id)" +
+                " values ('" + participantCohortStatus.getBirthDate().toString() + "', 1, 1, 1, 0, 1, 1)";
+
+        when(jdbcTemplate.getDataSource()).thenReturn(mockDatasource);
```
Comment (danqrodney) [inline api/src/test/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImplTest.java]: Is there another way to test this, using JDBC with an inmemory DB or something?

Having all these mocks doesn't really tell us whether this works...

File: api/src/main/java/org/pmiops/workbench/db/dao/ParticipantCohortStatusDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                    sqlStatement = SQL_TEMPLATE;
+                }
+            }
+
+            if (!sqlStatement.equals(SQL_TEMPLATE)) {
+                statement.execute(sqlStatement);
+            }
+
+            connection.commit();
+            connection.setAutoCommit(true);
```
Comment (danqrodney) [inline api/src/main/java/org/pmiops/workbench/db/dao/ParticipantCohortStatusDaoImpl.java]: I'm wondering if this should be in a finally block? Will setAutoCommit(false) above affect other pooled connections if this doesn't run?

Comment (freemabd1) [reply api/src/main/java/org/pmiops/workbench/db/dao/ParticipantCohortStatusDaoImpl.java]: Potentially. Probably better to be safe a move it into finally. I created a new PR for this: https://github.com/all-of-us/workbench/pull/421

**Pull request merged**